kwanj-k/ride-my-way-API-V1
# #158502023-Adds rides endpoints

### What does this PR do?
Adds rides related endpoints

### Description of Task to be completed
* create end-point to get all rides
* create end-point to get a specific ride by id
* create end-point to edit an existing ride by id
* create end-point to delete/cancel a ride offer

### How should I test it?
 1. Clone the repo
  ```
  $ git clone https://github.com/kwanj-k/ride-my-way-API-V1.git
  $ cd into the cloned repo
  $ git checkout ft-get-ride-#158502023-
  ```
Create a virtualenv and activate it::

    Virtualenv -p python3 .
    source bin/activate

Install the dependencies::

    pip install -r requirements.txt 

### Running tests without coverage
Run the tests from the terminal
```
python manage.py test
```

### Running tests with coverage
Run tests with coverage by running this command in the terminal
```
pytest --cov-report term-missing --cov=app
```
## Testing on Postman
Run the development server:
  ```
  $ flask run
  ```
Start Postman
Rides data:
```
{
"origin":"Thika",
"destination":"Nairobi",
"departure":"5.30am"
}
```
Using rides data:

* Make a `POST` request to `/api/v1/rides/`
* Make a `GET` request to `/api/v1/rides/`
* Make a `GET` request to `/api/v1/rides/id`
* Make a `PUT` request to `/api/v1/rides/id`
* Make a `DELETE` request to `/api/v1/rides/id`

All the end-points should respond accordingly


### What PT stories?
[ft-get-ride-#158502023]https://www.pivotaltracker.com/story/show/158502023

[#158502079]https://www.pivotaltracker.com/story/show/158502079

[#158502167]https://www.pivotaltracker.com/story/show/158502167
